src/NHibernate/Linq/Visitors/ExpressionParameterVisitor.cs

```diff
@@ -107,6 +102,17 @@ protected override Expression VisitMethodCall(MethodCallExpression expression)
 				return Expression.Call(null, expression.Method, query, arg);
 			}
 
+			if (_functionRegistry != null &&
+				_functionRegistry.TryGetGenerator(method, out var generator) &&
+				generator is CollectionContainsGenerator)
```

Comment:
What about CollectionContainsRuntimeHqlGenerator?

Reply:
Ok, I've understood how it works. CollectionContainsRuntimeHqlGenerator returns CollectionContainsGenerator for a mehtod. So this code works.